test/integration/fixtures/hooks/multiple-hook-async-error.fixture.js

```diff
@@ -1,141 +1,160 @@
+    runOrder.push('1 after each');
   });
-
-  after(function () {
-    console.log('1 after');
+  after('1 after', function () {
+    runOrder.push('1 after');
   });
 });
 
 describe('2', function () {
-  beforeEach(function (done) {
-    console.log('2 before each');
+  beforeEach('2 before each', function (done) {
+    runOrder.push('2 before each');
     process.nextTick(function () {
-      throw new Error('2 before each hook failed');
+      done(new Error('2 before each hook failed'));
     });
   });
 
   describe('2-1', function () {
-    before(function () {
-      console.log('2-1 before');
+    before('2-1 before', function () {
+      runOrder.push('2-1 before');
     });
-    beforeEach(function () {
-      console.log('2-1 before each');
+    beforeEach('2-1 before each', function () {
+      //  should not run
     });
+
     it('2-1 test 1', function () {
-      console.log('2-1 test 1');
+      //  should not run
     });
     it('2-1 test 2', function () {
-      console.log('2-1 test 2');
+      //  should not run
     });
-    afterEach(function () {
-      console.log('2-1 after each');
+
+    afterEach('2-1 after each', function () {
+      //  should not run
     });
-    after(function () {
-      console.log('2-1 after');
+    after('2-1 after', function () {
+      runOrder.push('2-1 after');
     });
   });
 
   describe('2-2', function () {
-    before(function () {
-      console.log('2-2 before');
+    before('2-2 before', function () {
+      runOrder.push('2-2 before');
     });
-    beforeEach(function () {
-      console.log('2-2 before each');
+    beforeEach('2-2 before each', function () {
+      runOrder.push('2-2 before each');
     });
+
     it('2-2 test 1', function () {
-      console.log('2-2 test 1');
+      //  should not run
     });
     it('2-2 test 2', function () {
-      console.log('2-2 test 2');
+      //  should not run
     });
-    afterEach(function () {
-      console.log('2-2 after each');
+
+    afterEach('2-2 after each', function () {
+      runOrder.push('2-2 after each');
     });
-    after(function () {
-      console.log('2-2 after');
+    after('2-2 after', function () {
+      runOrder.push('2-2 after');
     });
   });
 
-  afterEach(function (done) {
-    console.log('2 after each');
+  afterEach('2 after each', function (done) {
+    runOrder.push('2 after each');
     process.nextTick(function () {
-      throw new Error('2 after each hook failed');
+      done(new Error('2 after each hook failed'));
     });
   });
-
-  after(function () {
-    console.log('2 after');
+  after('2 after', function () {
+    runOrder.push('2 after');
   });
 });
 
-after(function () {
-  console.log('root after');
+afterEach('root after each', function () {
+  runOrder.push('root after each');
 });
-afterEach(function () {
-  console.log('root after each');
+after('root after', function () {
+  runOrder.push('root after');
+  assert.deepEqual(runOrder, [
```

Comment:
Do your assertion check in a separate `after`.

Reply:
Where did this go?

Reply:
I reversed my changes to this file, out of topic.